specs/waku/v2/waku-v2-rpc-api.md

```diff
@@ -0,0 +1,459 @@
+
+#### Parameters
+
+| Field | Type | Inclusion | Description |
+| ----: | :--: | :--: |----------- |
+| `topics` | `Array`[`Number`] | mandatory | Array of content topics to query for historical messages |
+| `pagingOptions` | [`PagingOptions`](#PagingOptions) | optional | Pagination information |
+
+#### Response
+
+- [**`StoreResponse`**](#StoreResponse) - the response to a `query` for historical messages.
+
+## Filter API
+
+Refer to the [Waku Filter specification](https://github.com/vacp2p/specs/blob/master/specs/waku/v2/waku-filter.md) for more information on content filtering.
+
+### Types
+
+The following structured types are defined for use on the Filter API:
+
+#### ContentFilter
+
+`ContentFilter` is an `Object` containing the following fields:
+| Field | Type | Inclusion | Description |
+| ----: | :--: | :--: |----------- |
+| `topics` | `Array`[`Number`] | mandatory | Array of message content topics |
+
+### `post_waku_v2_filter_v1_subscription`
+
+The `post_waku_v2_filter_v1_subscription` method creates a subscription in a [light node](https://github.com/vacp2p/specs/blob/master/specs/waku/v2/waku-filter.md#rationale) for messages that matches a content filter and, optionally, a [PubSub `topic`](https://github.com/libp2p/specs/blob/master/pubsub/README.md#the-topic-descriptor).
+
+#### Parameters
+
+| Field | Type | Inclusion | Description |
+| ----: | :--: | :--: |----------- |
+| `contentFilters` | `Array`[[`ContentFilter`](#ContentFilter)] | mandatory | Array of content filters being subscribed to |
+| `topic` | `String` | optional | Message topic |
+
+#### Response
+
+- **`Bool`** - `true` on success or an [error](https://www.jsonrpc.org/specification#error_object) on failure.
+
+### `delete_waku_v2_filter_v1_subscription`
+
+The `delete_waku_v2_filter_v1_subscription` method removes subscriptions in a [light node](https://github.com/vacp2p/specs/blob/master/specs/waku/v2/waku-filter.md#rationale) matching a content filter and, optionally, a [PubSub `topic`](https://github.com/libp2p/specs/blob/master/pubsub/README.md#the-topic-descriptor).
+
+#### Parameters
+
+| Field | Type | Inclusion | Description |
+| ----: | :--: | :--: |----------- |
+| `contentFilters` | `Array`[[`ContentFilter`](#ContentFilter)] | mandatory | Array of content filters being unsubscribed from |
+| `topic` | `String` | optional | Message topic |
+
+#### Response
+
+- **`Bool`** - `true` on success or an [error](https://www.jsonrpc.org/specification#error_object) on failure.
```

Comment:
For external links, it might be a good idea to also have Reference list at the end. Something like:

### References

1. [JSON RPC spec](www.jsonrpc.org/...)
2. [libp2p PubSub](...)

etc, just to clarify external dependencies.

Reply:
Thanks! Will do in subsequent PR.